### PR TITLE
mysql_user: fix idempotence check for ALL GRANT In MySQL 8.0 

### PR DESCRIPTION
**Pull request by azeliashchonak**
##### SUMMARY

In MySQL 8.0 SHOW GRANTS no longer displays ALL PRIVILEGES in its global-privileges output. Instead, SHOW GRANTS explicitly lists each granted global privilege. The issue is that the mysql_user module is using a simple comparison between new and current privileges and if we define "_._:ALL,GRANT" it will return changed, because "ALL" != list_of_privileges, so idempotence check will be failed.

##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

mysql_user
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/database/mysql/mysql_user.py
Comment (bmalynovytch) [inline lib/ansible/modules/database/mysql/mysql_user.py]: You'll need to rebase to fix the regex (see https://github.com/ansible/ansible/pull/52278/files#diff-399cf46d54197b26de359e13cb8bc033R430 )
```suggestion
            """GRANT (.+) ON (.+) TO (['`"]).*\\3@(['`"]).*\\4( IDENTIFIED BY PASSWORD (['`"]).+\\6)? ?(.*)""", grant[0])
```

**Review by bmalynovytch — CHANGES_REQUESTED** [review]

**Review by bmalynovytch — APPROVED** [review]

**Comment by Xyon** [comment]
Code LGTM, not tested however.

**Comment by azeliashchonak** [comment]
ready_for_review

**Comment by bmalynovytch** [comment]
@azeliashchonak I'm waiting for my own PR to be merged, because you'll need to rebase / rework it.
I'll review as soon as the merge + rebase will be ok

**Comment by azeliashchonak** [comment]
ready_for_review

**Comment by tgadiev** [comment]
@bmalynovytch can you pls take a look?

**Comment by bmalynovytch** [comment]
Still waiting for #45355 to be merged before.
We need more people testing and reviewing it before it gets merged.

**Comment by azeliashchonak** [comment]
Created new with clear code https://github.com/ansible/ansible/pull/55140

**Pull request closed without merging**
